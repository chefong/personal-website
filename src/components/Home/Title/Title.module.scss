.section {
  height: 100vh;
}

.subtitle {
  font-weight: bold;
  font-size: 2em;
  color: var(--theme-color);
}

.name {
  font-weight: bold;
  font-size: 6em;
  color: var(--theme-page-text);
  margin-left: -4px;
}

.description {
  color: var(--theme-page-text);
}

.period {
  color: var(--theme-color);
}

.button {
  font-weight: bold;
  padding: 10px 16px;
  border-radius: 8px;
}

.resume {
  background-color: var(--theme-page-text);
  border: none;
}

.resume:hover {
  background-color: var(--theme-color);
}

.contact {
  background: none;
  color: var(--theme-page-text);
  border: 2px solid var(--theme-page-text);
}

.contact:hover {
  color: var(--theme-color);
  border-color: var(--theme-color);
  background: none;
}

.resume {
  border: 2px solid var(--theme-page-text);
  color: var(--theme-page-background);
}

.resume:hover {
  background-color: var(--theme-color);
  border-color: var(--theme-color);
}

.contactIcon, .resumeIcon {
  width: 0.9em;
  margin-right: 10px;
  margin-top: -2px;
}

.hero {
  width: 95%;
  display: block;
  margin: 0 auto;
  margin-top: 1.5em;
  transition: all 0.25s ease;
  filter: saturate(0);
  border-radius: 50%;
  border: 8px solid var(--theme-color);
}

.hero:hover {
  filter: saturate(1);
}

@media (max-width: 768px) {
  .hero {
    width: 60%
  }
}

@media (max-width: 576px) {
  .section {
    padding-top: 4em;
  }

  .hero {
    width: 75%
  }
}

@media (max-width: 320px) {
  .section {
    padding-top: 6em;
  }

  .resume {
    margin-top: 8px;
  }
}

.particles {
  position: absolute;
  background-size: cover;
  max-width: 1600px;
}
