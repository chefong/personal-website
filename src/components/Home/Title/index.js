import { useState, useContext } from 'react';
// import { Button } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FormattedMessage } from 'text-provider';
import Confetti from 'react-dom-confetti';
import Particles from 'react-particles-js';
import styles from './styles.module.scss';
import { store } from '~/store/GlobalProvider';
import Button from '~/components/Button';
import { gaEvents, particlesParams } from '~/common/constants';
import { logToGA } from '~/common/utils/ga';

const confettiConfig = {
    angle: 90,
    spread: '80',
    startVelocity: 35,
    elementCount: 30,
    dragFriction: 0.12,
    duration: 2500,
    stagger: 5,
    width: '10px',
    height: '10px',
    perspective: '500px',
    colors: ['#a864fd', '#29cdff', '#78ff44', '#ff718d', '#fdff6a'],
};

export default function Title() {
    const [resumeHovered, setResumeHovered] = useState(false);
    const { state } = useContext(store);
    const { theme } = state;

    const handleResumeHoverIn = (e) => {
        e.preventDefault();
        setResumeHovered(true);
    };

    const handleResumeHoverOut = (e) => {
        e.preventDefault();
        setResumeHovered(false);
    };

    return (
        <section
            className={`${styles.section} container-fluid`}
            data-testid="Title-container"
        >
            {theme === 'dark' && (
                <Particles
                    canvasClassName={styles.particles}
                    params={particlesParams}
                />
            )}
            <div className="h-100 row justify-content-center align-items-center">
                <div className="col-lg-8 col-md-6">
                    <div>
                        <p className={styles.subtitle}>
                            <FormattedMessage id="TITLE_GREETING" />
                        </p>
                        <h1 className={styles.name}>
                            <FormattedMessage id="NAME" />
                            <span className={styles.period}>.</span>
                        </h1>
                        <p className={styles.description}>
                            <FormattedMessage id="TITLE_DESCRIPTION" />
                        </p>
                        <div className="mt-5">
                            <a
                                href="mailto:ericong18@gmail.com"
                                className="mr-4 mb-4"
                                onClick={() =>
                                    logToGA({
                                        action: gaEvents.CONTACT_ME_CLICKED,
                                    })
                                }
                            >
                                <Button variation="outline">
                                    <FontAwesomeIcon
                                        className={styles.contactIcon}
                                        icon="paper-plane"
                                    />
                                    <FormattedMessage id="TITLE_CONTACT" />
                                </Button>
                            </a>
                            <a
                                href="https://drive.google.com/file/d/1XdYGfHC1Irt0oh5YwNBZSWQhXAxFNGnL/view?usp=sharing"
                                onMouseEnter={handleResumeHoverIn}
                                onMouseLeave={handleResumeHoverOut}
                                onClick={() =>
                                    logToGA({ action: gaEvents.RESUME_CLICKED })
                                }
                                className="mr-4 mb-4"
                            >
                                <Button variation="primary">
                                    <FontAwesomeIcon
                                        className={styles.resumeIcon}
                                        icon="file-alt"
                                    />
                                    <FormattedMessage id="TITLE_RESUME" />
                                    <Confetti
                                        active={resumeHovered}
                                        config={confettiConfig}
                                    />
                                </Button>
                            </a>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    );
}
