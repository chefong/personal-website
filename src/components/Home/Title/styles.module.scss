.section {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}

.subtitle {
  font-weight: 800;
  font-size: 32px;
  color: var(--theme-color);
  margin: 0;
}

.name {
  font-weight: 800;
  font-size: 96px;
  color: var(--theme-page-text);
  margin: 0;
}

.description {
  color: var(--theme-page-text);
  margin: 0;
}

.period {
  color: var(--theme-color);
}

.content {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.buttonGroup {
  display: flex;
  gap: 16px;
  margin: 48px 0 0 0;
}

.link:hover {
  text-decoration: none;
}

.icon {
  width: 16px;
}

.hero {
  width: 95%;
  display: block;
  margin: 0 auto;
  margin-top: 1.5em;
  transition: all 0.25s ease;
  filter: saturate(0);
  border-radius: 50%;
  border: 8px solid var(--theme-color);
}

.hero:hover {
  filter: saturate(1);
}

@media (max-width: 768px) {
  .hero {
    width: 60%
  }
}

@media (max-width: 576px) {
  .section {
    padding-top: 4em;
  }

  .hero {
    width: 75%
  }
}

@media (max-width: 320px) {
  .section {
    padding-top: 6em;
  }

  .resume {
    margin-top: 8px;
  }
}

.particles {
  position: absolute;
  background-size: cover;
  max-width: 1600px;
}
