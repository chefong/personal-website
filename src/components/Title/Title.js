import { useState } from 'react';
import { Button } from 'reactstrap';
import styles from './Title.module.css';

const darkPlaneIconPath = "/imgs/title/paper-plane-solid.svg";
const lightPlaneIconPath = "/imgs/title/paper-plane-solid-light.svg";

const bwProfileImagePath = "/imgs/title/me.svg";
const coloredProfileImagePath = "/imgs/title/me-colored.svg";

export default function Title() {
  const [contactIconPath, setContactIconPath] = useState(darkPlaneIconPath);
  const [profileImagePath, setProfileImagePath] = useState(bwProfileImagePath);

  const handleContactHoverIn = e => {
    e.preventDefault();
    setContactIconPath(lightPlaneIconPath);
  }

  const handleContactHoverOut = e => {
    e.preventDefault();
    setContactIconPath(darkPlaneIconPath);
  }

  const handleProfileHoverIn = e => {
    e.preventDefault();
    setProfileImagePath(coloredProfileImagePath);
  }

  const handleProfileHoverOut = e => {
    e.preventDefault();
    setProfileImagePath(bwProfileImagePath);
  }

  return (
    <section className={`${styles.section} container-fluid`}>
      <div className="row justify-content-center">
        <div className="col-md-4">
          <div className={`${styles.content} mb-5`}>
            <p className={styles.subtitle}>Hi, This Is</p>
            <h1 className={styles.name}>Eric Ong<span className={styles.period}>.</span></h1>
            <p className={styles.description}>Iâ€™m a Software Engineer based in the SF Bay Area with a focus on full-stack web development.</p>
            <div className="mt-5">
              <a href="mailto:ericong18@gmail.com">
                <Button className={`${styles.button} ${styles.contact} mr-4`} onMouseEnter={handleContactHoverIn} onMouseLeave={handleContactHoverOut}>
                  <img className={styles.resumeIcon} src={contactIconPath} alt="" />
                  Contact Me
                </Button>
              </a>
              <a href="https://drive.google.com/file/d/1p9soqZuF9Ud7QaSsN0SV9fIjAScFseV3/view?usp=sharing">
                <Button className={`${styles.button} ${styles.resume} mr-4`}>
                  <img className={styles.resumeIcon} src="/imgs/title/file-solid.svg" alt=""/>
                  Resume
                </Button>
              </a>
            </div>
          </div>
        </div>
        <div className="col-md-3 offset-md-1">
          <img
            className={`${styles.hero}`}
            src={profileImagePath}
            alt="Eric Ong"
            onMouseEnter={handleProfileHoverIn}
            onMouseLeave={handleProfileHoverOut}
          />
        </div>
      </div>
    </section>
  )
}