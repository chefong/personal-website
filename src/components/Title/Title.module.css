.section {
  margin-top: 10em;
}

.content {
  margin-bottom: 2em;
}

.subtitle {
  font-weight: 500;
  font-size: 2em;
  color: #4A74E1;
}

.name {
  font-weight: 500;
  font-size: 6em;
  color: #303030;
  margin-left: -4px;
}

.description {
  color: #303030;
}

.period {
  color: #4A74E1;
}

.button {
  font-weight: 500;
  padding: 10px 16px;
  border-radius: 8px;
}

.resume {
  background-color: #303030;
}

.resume:hover {
  background-color: #404040;
}

.contact {
  background: none;
  color: #303030;
  border: 2px solid #303030;
}

.contact:hover {
  background-color: #303030;
  border: 2px solid #303030;
  color: #FFFEFB;
}

.contactIcon, .resumeIcon {
  width: 0.9em;
  margin-right: 8px;
  margin-top: -2px;
}

.hero {
  width: 100%;
  display: block;
  margin-left: auto;
  margin-top: 1em;
  transition: all 0.25s ease;
}