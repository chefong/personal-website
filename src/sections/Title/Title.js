import { useState } from 'react';
import { Button } from 'reactstrap';
import { Fade } from 'react-reveal';
import { revealProps } from '../../shared/constants';
import styles from './Title.module.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Confetti from 'react-dom-confetti';

const confettiConfig = {
  angle: 90,
  spread: "80",
  startVelocity: 35,
  elementCount: 50,
  dragFriction: 0.12,
  duration: 2500,
  stagger: 5,
  width: "10px",
  height: "10px",
  perspective: "500px",
  colors: ["#a864fd", "#29cdff", "#78ff44", "#ff718d", "#fdff6a"]
};

export default function Title() {
  const [profileIn, setProfileIn] = useState(false);
  const [resumeHovered, setResumeHovered] = useState(false);

  const handleProfileHoverIn = e => {
    e.preventDefault();
    setProfileIn(true);
  }

  const handleProfileHoverOut = e => {
    e.preventDefault();
    setProfileIn(false);
  }

  const handleResumeHoverIn = e => {
    e.preventDefault();
    setResumeHovered(true);
  }

  const handleResumeHoverOut = e => {
    e.preventDefault();
    setResumeHovered(false);
  }

  return (
    <section className={`${styles.section} container-fluid`}>
      <div className="row justify-content-center">
        <div className="col-lg-4 col-md-6">
          <Fade {...revealProps}>
            <div className={`${styles.content} mb-5`}>
              <p className={styles.subtitle}>Hi, This Is</p>
              <h1 className={styles.name}>Eric Ong<span className={styles.period}>.</span></h1>
              <p className={styles.description}>Iâ€™m a Software Engineer based in the SF Bay Area with a focus on full-stack web development.</p>
              <div className="mt-5">
                <a href="mailto:ericong18@gmail.com" className="mr-4 mb-4">
                  <Button className={`${styles.button} ${styles.contact}`}>
                    <FontAwesomeIcon className={styles.contactIcon} icon="paper-plane" />
                    Contact Me
                  </Button>
                </a>
                <a
                  href="/docs/EricOngResume.pdf"
                  onMouseEnter={handleResumeHoverIn}
                  onMouseLeave={handleResumeHoverOut}
                  className="mr-4 mb-4" 
                >
                  <Button className={`${styles.button} ${styles.resume}`}>
                    <FontAwesomeIcon className={styles.resumeIcon} icon="file-alt" />
                    Resume
                    <Confetti active={resumeHovered} config={confettiConfig} />
                  </Button>
                </a>
              </div>
            </div>
          </Fade>
        </div>
        <div className="col-lg-3 offset-lg-1 col-md-4">
          <Fade {...revealProps}>
            <img
              className={styles.hero}
              src="/imgs/title/me.png"
              alt="Eric Ong"
              onMouseEnter={handleProfileHoverIn}
              onMouseLeave={handleProfileHoverOut}
              style={{ filter: profileIn ? "saturate(1)" : "saturate(0)" }}
            />
          </Fade>
        </div>
      </div>
    </section>
  )
}