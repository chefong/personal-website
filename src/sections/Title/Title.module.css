.section {
  margin-top: 10em;
}

.content {
  margin-bottom: 2em;
}

.subtitle {
  font-weight: 500;
  font-size: 2em;
  color: #4A74E1;
}

.name {
  font-weight: 500;
  font-size: 6em;
  color: #303030;
  margin-left: -4px;
}

.description {
  color: #303030;
}

.period {
  color: #4A74E1;
}

.button {
  font-weight: 500;
  padding: 10px 16px;
  border-radius: 8px;
}

.resume {
  background-color: #303030;
  border: none;
}

.resume:hover {
  background-color: #4A74E1;
}

.contact {
  background: none;
  color: #303030;
  border: 2px solid #303030;
}

.contact:hover {
  color: #4A74E1;
  border: 2px solid #4A74E1;
  background: none;
}

.contactIcon, .resumeIcon {
  width: 0.9em;
  margin-right: 10px;
  margin-top: -2px;
}

.hero {
  width: 95%;
  display: block;
  margin: 0 auto;
  margin-top: 1.5em;
  transition: all 0.25s ease;
  filter: saturate(0);
  border-radius: 50%;
  border: 8px solid #4A74E1;
}

.hero:hover {
  filter: saturate(1);
}