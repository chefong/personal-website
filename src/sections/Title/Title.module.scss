.section {
  padding-top: 16em;
  margin-bottom: 14em;
}

.content {
  margin-bottom: 2em;
}

.subtitle {
  font-weight: 500;
  font-size: 2em;
  color: var(--theme-color);
}

.name {
  font-weight: 500;
  font-size: 6em;
  color: var(--theme-page-text);
  margin-left: -4px;
}

.description {
  color: var(--theme-page-text);
}

.period {
  color: var(--theme-color);
}

.button {
  font-weight: 500;
  padding: 10px 16px;
  border-radius: 8px;
}

.resume {
  background-color: var(--theme-page-text);
  border: none;
}

.resume:hover {
  background-color: var(--theme-color);
}

.contact {
  background: none;
  color: var(--theme-page-text);
  border: 2px solid var(--theme-page-text);
}

.contact:hover {
  color: var(--theme-color);
  border-color: var(--theme-color);
  background: none;
}

.resume {
  border: 2px solid var(--theme-page-text);
  color: var(--theme-page-background);
}

.resume:hover {
  background-color: var(--theme-color);
  border-color: var(--theme-color);
}

.contactIcon, .resumeIcon {
  width: 0.9em;
  margin-right: 10px;
  margin-top: -2px;
}

.hero {
  width: 95%;
  display: block;
  margin: 0 auto;
  margin-top: 1.5em;
  transition: all 0.25s ease;
  filter: saturate(0);
  border-radius: 50%;
  border: 8px solid var(--theme-color);
}

.hero:hover {
  filter: saturate(1);
}

@media (max-width: 767px) {
  .hero {
    width: 60%
  }
}

@media (max-width: 576px) {
  .hero {
    width: 75%
  }
}

@media (max-width: 576px) {
  .section {
    padding-top: 8em;
  }

  .hero {
    width: 75%
  }
}
