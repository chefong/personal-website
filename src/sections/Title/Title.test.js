import React from 'react';
import { render } from '@testing-library/react';
import Title from './Title';

describe('Title', () => {
  it('renders', () => {
    const { getByTestId } = render(<Title />);
    expect(getByTestId('Title-container')).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(<Title />);
    expect(container).toMatchSnapshot();
  });
});
