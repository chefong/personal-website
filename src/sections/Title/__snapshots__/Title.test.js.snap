// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches snapshot 1`] = `
<div>
  <section
    class="undefined container-fluid"
    data-testid="Title-container"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-lg-4 col-md-6"
      >
        <div
          class="react-reveal undefined mb-5"
        >
          <p>
            <span />
          </p>
          <h1>
            <span />
            <span>
              .
            </span>
          </h1>
          <p>
            <span />
          </p>
          <div
            class="mt-5"
          >
            <a
              class="mr-4 mb-4"
              href="mailto:ericong18@gmail.com"
            >
              <button
                class="undefined undefined btn btn-secondary"
              >
                <span />
              </button>
            </a>
            <a
              class="mr-4 mb-4"
              href="/docs/EricOngResume.pdf"
            >
              <button
                class="undefined undefined btn btn-secondary"
              >
                <span />
                <div
                  style="position: relative;"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-lg-3 offset-lg-1 col-md-4"
      >
        <img
          alt="Eric Ong"
          src="/imgs/title/me.jpg"
        />
      </div>
    </div>
  </section>
</div>
`;
